ArduPlane/quadplane.cpp

```diff
@@ -1178,10 +1187,28 @@ void QuadPlane::control_loiter()
     // run loiter controller
     loiter_nav->update();
 
-    // nav roll and pitch are controller by loiter controller
+    // nav roll and pitch are controlled by loiter controller
     plane.nav_roll_cd = loiter_nav->get_roll();
     plane.nav_pitch_cd = loiter_nav->get_pitch();
 
+    // avoid flying backwards at high speed
+    Vector3f vel_ned;
+    float airspeed = 0.0f;
+    if (ahrs.airspeed_estimate(&airspeed) && plane.ahrs.get_velocity_NED(vel_ned)) {
+        Vector3f vel_body = ahrs.get_rotation_body_to_ned().transposed() * vel_ned;
+
+        // handle negative effective airspeed
+        float airspeed_eff = vel_body.x + airspeed;
+        if (airspeed_eff < 0.0f && (plane.channel_pitch->get_control_in() > 0)) {
+            // ramp pitch down to zero as reverse speed increases
+            plane.nav_pitch_cd *= 1.0f + (airspeed_eff / loit_rev_spd);
+            plane.nav_pitch_cd = constrain_float(plane.nav_pitch_cd,
+                                                 -loiter_nav->get_angle_max_cd(),
+                                                 loiter_nav->get_angle_max_cd());
+            loiter_nav->init_target();
```

Comment:
You can't do this after update on loiter nav. This resets the desired velocity/acceleration to be the current vehicle velocity/acceleration.

Reply:
I've already overridden the loiter pitch target which was set by loiter_nav->update()
The call to init_target effectively relaxes the position controller, but perhaps there is a more elegant way to do that?

Reply:
I'm multitasking my review with a call, so I may be missing something, but isn't this effectively relaxing the controller continuously while trying to fly backwards in a no wind setup?

Reply:
What I was trying to achieve was to override the loiter controller with a pitch controller which limits airspeed to -loit_rev_spd (but only when the pilot is applying up elevator and airspeed is negative).

It behaves well in RF8/SITL, but I haven't flight tested it yet.  I will flight test on a Convergence tiltrotor but am all out of standard quadplanes :)